cpp/log/cert_checker.cc

```diff
@@ -178,12 +179,15 @@ Status CertChecker::CheckIssuerChain(CertChain* chain) const {
 Status CertChecker::CheckPreCertChain(PreCertChain* chain,
                                       string* issuer_key_hash,
                                       string* tbs_certificate) const {
-  if (!chain || !chain->IsLoaded())
+  if (!chain || !chain->IsLoaded()) {
     return Status(util::error::INVALID_ARGUMENT, "invalid certificate chain");
-  Cert::Status status = chain->IsWellFormed();
-  if (status == Cert::FALSE)
+  }
+
+  StatusOr<bool> status = chain->IsWellFormed();
+  if (status.ok() && !status.ValueOrDie()) {
     return Status(util::error::INVALID_ARGUMENT, "prechain not well formed");
-  if (status != Cert::TRUE) {
+  }
+  if (!status.ok() || !status.ValueOrDie()) {
```

Comment:
I think you're right. Fixing it now and running the tests again.

On 17 September 2015 at 15:59, Al Cutter <notifications@github.com> wrote:

> In cpp/log/cert_checker.cc
> <https://github.com/google/certificate-transparency/pull/922#discussion_r39755620>
> :
>
> >      return Status(util::error::INVALID_ARGUMENT, "prechain not well formed");
> > -  if (status != Cert::TRUE) {
> > +  }
> > +  if (!status.ok() || !status.ValueOrDie()) {
>
> I think it's right; if chain_well_formed was false (and ok), then we'd
> have returned at that point, so at 190 we only have to check if there was
> an error with the !ok().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/922/files#r39755620>
> .
>